src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets

```diff
@@ -20,18 +20,19 @@ Copyright (c) .NET Foundation. All rights reserved.
     <MicrosoftNETBuildTasksTFM Condition=" '$(MicrosoftNETBuildTasksTFM)' == ''">net46</MicrosoftNETBuildTasksTFM>
     <MicrosoftNETBuildTasksDirectory>$(MicrosoftNETBuildTasksDirectoryRoot)$(MicrosoftNETBuildTasksTFM)/</MicrosoftNETBuildTasksDirectory>
     <MicrosoftNETBuildTasksAssembly>$(MicrosoftNETBuildTasksDirectory)Microsoft.NET.Build.Tasks.dll</MicrosoftNETBuildTasksAssembly>
-    
-    <!-- 
+    <IsExecutable Condition="'$(IsExecutable)' == '' and '$(OutputType)'=='Exe'">true</IsExecutable>
+    <IsExecutable Condition="'$(IsExecutable)' == '' and '$(OutputType)'=='WinExe'">true</IsExecutable>
```

Comment:
I believe this is after the use in BeforeCommon.targets and so the change won't have the effect you want. 

Mind you, I'm surprised this test didn't fail: https://github.com/dotnet/sdk/blob/1d81050cab00c6f9ab8bf5b984bc42eff2760c1e/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs#L351 

I'm surprised to see it doesn't check the contents of the app.config. Can you check  that before and after your change?